platform/darwin/src/MGLTypes.h

```diff
@@ -110,11 +115,13 @@ NS_INLINE NSString *MGLStringFromMGLTransition(MGLTransition transition) {
  animation.
  
  @return Returns a `MGLTransition` struct containing the transition attributes.
+ The default value for `enablePlacementTransitions` is `YES`.
  */
 NS_INLINE MGLTransition MGLTransitionMake(NSTimeInterval duration, NSTimeInterval delay) {
     MGLTransition transition;
     transition.duration = duration;
     transition.delay = delay;
+    transition.enablePlacementTransitions = YES;
```

Comment:
Setting this field to anything other than `NO` is a potential trap for developers: factory `*Make()` methods are supposed to mimic the behavior of initializer argument lists. Developers may reasonably expect all three of the following lines to behave identically in Objective-C and Objective-C++:

```objc
MGLTransition transition = MGLTransitionMake(1, 2);
MGLTransition transition = (MGLTransitionMake) {1, 2};
MGLTransition transition = (MGLTransitionMake) { .duration = 1, .delay = 2 };
```

(`*Make()` functions are discouraged in Swift in favor of initializers that are translated under the hood to initializer argument lists.)

Since adding an explicit `enablePlacementTransitions` argument to `MGLTransitionMake()` would be a backwards-incompatible change, I think our only choice would be to leave out this line and document that developers should set `enablePlacementTransitions` to `YES` after calling `MGLTransitionMake()`. That’s still quite an easy trap to fall into, so we should look into higher-level solutions like a transition class or a separate property.